### ISSUE TITLE
Make on sale prices accessible

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-gutenberg-products-block#1668]
In many blocks we display on sale products with the old price wrapped by `<del>` tags and the current price next to it:

![imatge](https://user-images.githubusercontent.com/3616980/73348579-eea58880-4289-11ea-8efe-b69c556584b8.png)
![imatge](https://user-images.githubusercontent.com/3616980/73348530-de8da900-4289-11ea-800c-21c7cf4ed5d0.png)
![imatge](https://user-images.githubusercontent.com/3616980/73348744-388e6e80-428a-11ea-8f00-5da42f1a90ec.png)

Currently, screen readers read both prices without specifying which one is the old one and which one is the current one.

We need a solution for that. It should work for JS-rendered blocks and PHP-rendered.

### Blocks fixed
* [ ] Featured Product
* [ ] Featured Category
* [ ] Product grid blocks
* [x] All Products (#3065)
* [x] Single Product (#3065)
* [x] Product Price (#3065)
* [x] Cart (#2041)
* [x] Checkout (#2041)

**Comment by Aljullu** [comment]
In #2041 we made the _Cart_ and _Checkout_ product prices accessible. That work can be used as a reference: 2a7813d66de272d1f5502f2facf217be054e065a.

**Comment by gigitux** [comment]
After some investigation, I have some updates about the remaining blocks:

- `Product grid blocks` block is already fixed because it uses the `ProductPrice` component. This component already has the fix for accessibility. [Check this commit](https://github.com/woocommerce/woocommerce-gutenberg-products-block/commit/2a7813d66de272d1f5502f2facf217be054e065a)

- `Featured Category` block: it seems that there is no price reference in the source code. I was also not able to see the price in the UI.
![image](https://user-images.githubusercontent.com/4463174/140542361-76c1ee65-e34f-489f-a9eb-2be5039a1640.png)

– `Featured Product` block use on both side WP core functions. For the frontend side [we use ](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/e774294968ef6e270a46e43ccd9ac1055cd7c0b6/src/BlockTypes/FeaturedProduct.php#L68)`get_price_html() function` and for the editor side the price [is rendered by WC Core code ](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/e774294968ef6e270a46e43ccd9ac1055cd7c0b6/assets/js/blocks/featured-product/block.js#L331). After an internal discussion we decide to create a PR directly in WC Core in the future. (internal discussion p1636119978278700-slack-C8X6Q7XQU)


@Aljullu If you can confirm my conclusions, I propose to close this ticket once the WC ticket is created. What do you think?

**Comment by Aljullu** [comment]
> - `Product grid blocks` block is already fixed because it uses the `ProductPrice` component. This component already has the fix for accessibility. [Check this commit](https://github.com/woocommerce/woocommerce-gutenberg-products-block/commit/2a7813d66de272d1f5502f2facf217be054e065a)

Just to clarify, because the issue description might be confusing. We usually name 'Product grid blocks' a group of blocks that share a lot of code and which display a product grid. These are:

* Hand-picked Products Block
* Best Selling Products Block
* Top Rated Products Block
* Newest Products Block
* On Sale Products Block
* Products by Category Block
* Products by Tag Block
* Products by Attribute Block

(I know it might be a bit confusing at the beginning :sweat_smile: )

The render function of those blocks can be found in [`AbstractProductGrid.php`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/711d2fba27da0cf71c8597663aebc240e73407ce/src/BlockTypes/AbstractProductGrid.php), so it's all in PHP.

That means that those blocks aren't using the React component but instead, they are completely PHP-rendered. The code that renders the price seems to be located here:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/711d2fba27da0cf71c8597663aebc240e73407ce/src/BlockTypes/AbstractProductGrid.php#L539

So the price in those blocks seems to be rendered using the same method as the Featured Product block, and it should probably be fixed in WC core.

> - `Featured Category` block: it seems that there is no price reference in the source code. I was also not able to see the price in the UI.

Ah, good catch! Not sure why I listed it. :man_facepalming: 

> @Aljullu If you can confirm my conclusions, I propose to close this ticket once the WC ticket is created. What do you think?

Sounds good to me, thanks for taking care of this @gigitux!

**Comment by gigitux** [comment]
I opened [the issue on WooCommerce repo](https://github.com/woocommerce/woocommerce/issues/31099). I'm closing this issue.
